src/rust/engine/process_execution/src/lib.rs

```diff
@@ -150,9 +203,17 @@ impl AddAssign<UploadSummary> for ExecutionStats {
 pub trait CommandRunner: Send + Sync {
   fn run(
     &self,
-    req: ExecuteProcessRequest,
+    req: MultiPlatformExecuteProcessRequest,
     workunit_store: WorkUnitStore,
   ) -> BoxFuture<FallibleExecuteProcessResult, String>;
+  fn is_compatible_request(&self, req: &MultiPlatformExecuteProcessRequest) -> bool;
```

Comment:
Rather than having both `is_compatible_request` and `get_compatible_request`, I'd suggest having `get_compatible_request` return an `Option<ExecuteProcessRequest>` in order to do both jobs.

Alternatively, since you're exposing `get_platform`, would it be cleaner to have something like `get_compatible_platforms() -> &[Platform]`, and then have the caller check whether a request is compatible before calling a runner?

Finally, when `run` was the only method in the trait, it didn't need a docstring, because the API was more self-explanatory... with more methods here, they should definitely all have docstrings.

Reply:
Using an Option makes sense to me. I need to audit but its possible get_platform isn't needed.